Build/Build.cs

```diff
@@ -258,11 +275,38 @@ class Build : NukeBuild
         });
 
     Target SpellCheck => _ => _
+        .OnlyWhenDynamic(() => HasDocumentationChanges)
         .Executes(() =>
         {
             Node($"{YarnCli} install --silent", workingDirectory: RootDirectory);
             Node($"{YarnCli} --silent run cspell --no-summary", workingDirectory: RootDirectory,
                 customLogger: (_, msg) => Error(msg));
         });
+
+    bool HasDocumentationChanges =>
+        Changes.Any(x => x.StartsWith("docs"));
+
+    bool HasSourceChanges =>
+        Changes.Any(x => !x.StartsWith("docs"));
+
+    string[] Changes
+    {
+        get
+        {
+            using var repo = new Repository(GitRepository.LocalDirectory);
+
+            string baseRef = BaseRef ?? "develop";
```

Comment:
We have to possibilities here:

1) This solution
2) force the user to add the `BaseRef` if running local spell check `./build.sh SpellCheck --base-ref (develop|master|main|whatever)`

Reply:
How do you usually create a release? 

Do you
- locally merge `develop` into `master`, set a tag and push `master`
or
- creating a pull request from `develop` to `master`, set a tag and merge online?

I am asking because I have the feelings that this `baseRef`-thing could cause a little side effect, where all NUKE targets are skipped and no release is created.

Reply:
> How do you usually create a release?

I've done both and it shouldn't matter. I think that this `BaseRef` parameter should be optional. If you don't specify anything, just assume we want to run all targets. 

In fact, maybe we should rename that property to `PullRequestBase` or something like that.

Reply:
Ok makes sense

Yeah... If a variable name also confuses the author.. you should probably give it a better name 🙈